plugins/lookup/get_moid.py

```diff
@@ -0,0 +1,579 @@
+description:
+    - Returns Managed Object Reference (MoID) of the object contained in the specified path.
+options:
+    _terms:
+        type: dict
+        required: true
+        suboptions:
+            object_type:
+                description:
+                    - Type of the vSphere object to be queried
+                required: true
+                type: str
+            path:
+                description:
+                    - Type of the vSphere object to be queried
+                required: true
+                type:str
+    vcenter_hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_HOST) will be used instead.
+        required: true
+        type: str
+    vcenter_password:
+        description:
+            - The vSphere vCenter password.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_PASSWORD) will be used instead.
+        required: true
+        type: str
+    vcenter_rest_log_file:
+        description:
+            - You can use this optional parameter to set the location of a log file.
+            - This file will be used to record the HTTP REST interaction.
+            - The file will be stored on the host that run the module.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_REST_LOG_FILE) will be used instead.
+        type: str
+    vcenter_username:
+        description:
+            - The vSphere vCenter username.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_USER) will be used instead.
+        required: true
+        type: str
+    vcenter_validate_certs:
+        default: true
+        description:
+            - Allows connection when SSL certificates are not valid. Set to C(false) when
+              certificates are not trusted.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_VALIDATE_CERTS) will be used instead.
+        type: bool
+author:
+    - Alina Buzachis <@alinabuzachis>
+version_added: 1.0.1
+requirements:
+    - vSphere 7.0.2 or greater
+    - python >= 3.6
+    - aiohttp
+'''
+
+
+EXAMPLES = r'''
+# lookup sample
+
+- name: set connection info
+  set_fact:
+    connection_args:
+        vcenter_hostname: "vcenter.test"
+        vcenter_username: "administrator@vsphere.local"
+        vcenter_password: "1234"
+
+- name: lookup MoID of the object
+  debug: msg="{{ lookup('get_moid',
+                        {'vm', '/my_dc/host/my_cluster/esxi1.test/test_vm1'},
+                        **connection_args)}}"
+
+- name: lookup MoID of the object in the path
+  debug: msg="{{ lookup('get_moid', {'vm', '/my_dc/host/my_cluster/esxi1.test/'}) }}"
```

Comment:
I just tried: `lookup('vmware.vmware_rest.get_moid', {'cluster', '/my_dc/host/my_cluster'})`

and I get:
`
"msg": "template error while templating string: expected token ':', got ','. String: folder: {{ lookup('vmware.vmware_rest.get_moid', {'cluster', '/my_dc/host/my_cluster'}) }}"
`

Reply:
Sorry, that's my fault, it should be `"{{ lookup('get_moid', {'vm' : '/my_dc/host/my_cluster/esxi1.test/'}) }}"`.

Reply:
I'll rectify this in the documentation.